classes/mender-sdimg.bbclass

```diff
@@ -149,7 +149,7 @@ IMAGE_CMD_sdimg() {
     ROOTFS_SIZE=$(expr $REMAINING_SIZE / 2)
 
     # create rootfs
-    dd if=/dev/zero of=${WORKDIR}/rootfs.$FSTYPE count=1 seek=$ROOTFS_SIZE bs=1M
+    dd if=/dev/zero of=${WORKDIR}/rootfs.$FSTYPE count=0 seek=$ROOTFS_SIZE bs=1M
```

Comment:
IIRC mkfs.ext* would fail to create a FS if the whole image is sparse file. Have you verified that the rootfs is mountable?